src/Php/PolyfillPackagesProvider.php

```diff
@@ -50,7 +52,7 @@ public function provide(): array
      */
     private function filterPolyfillPackages(array $require): array
     {
-        return array_filter($require, static fn (string $packageName): bool => ! str_starts_with(
+        return array_filter(array_keys($require), static fn (string $packageName): bool => str_starts_with(
```

Comment:
here only `key` is needed, when we define in `composer.json`

```php
"require": {
    "symfony/polyfill-php80" => "*"
}
```

then, filter the `str_starts_with()` one instead, so it get data:

```php
array('symfony/polyfill-php80')
```

Reply:
I created e2e test for it:

- https://github.com/rectorphp/rector-src/pull/5392